ehitchhiking-ui/src/app/pages/main-screen/main-screen-component/main-screen.component.ts

```diff
@@ -164,18 +166,20 @@ export class MainScreenComponent implements OnInit {
 		this.activePassengerButton = true;
 		this.activeDriverButton = false;
 		this.userState = UserState.Passenger;
-    this.redrawPassengerDriverIconTriggers = {reset: true};
+		this.redrawPassengerDriverIconTriggers = {reset: true};
 		this.toggleMapInterfaceToDefault();
 	}
 
-	toggleStateToDriver() {
-		if (this.currentUser.cars.length !== 0) {
-			this.activeDriverButton = true;
-			this.activePassengerButton = false;
-			this.userState = UserState.Driver;
-			this.redrawPassengerDriverIconTriggers = {reset: true};
-			this.toggleMapInterfaceToDefault();
+	toggleStateToDriver(): void {
+		if (this.currentUser.cars.length === 0) {
```

Comment:
!this.currentUser || !this.currentUser.cars || !this.currentUser.cars.length

Reply:
fixed